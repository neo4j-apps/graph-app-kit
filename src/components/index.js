export { AsciiTable } from "./AsciiTable";
export { Chart } from "./Chart";
export { Cypher } from "./Cypher";
export { DesktopIntegration } from "./DesktopIntegration";
export { DriverProvider } from "./DriverProvider";
export { CypherEditor } from "./Editor";
export { GraphAppBase } from "./GraphAppBase";
export { Render } from "./Render";
