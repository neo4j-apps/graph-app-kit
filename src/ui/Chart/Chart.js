import React from "react";
import * as PropTypes from "prop-types";
import CypherChartWrapper from "./wrappers/CypherChartWrapper";
import ChartSwitch from "./wrappers/ChartSwitch";

export const Chart = props => {
  if (!props.type) return "No chart type set";
  switch (props.type) {
    case "cypher":
      return <CypherChartWrapper {...props} />;
    case "json":
      return <ChartSwitch {...props} />;
    default:
      return `${props.type} is an invalid chart type`;
  }
};

Chart.propTypes = {
  /** Data source. */
  type: PropTypes.oneOf(["cypher", "json"]).isRequired,
  /** Type of chart to render. */
  chartType: PropTypes.oneOf(["area", "bar", "text", "pie", "doughnut"])
    .isRequired,
  /** Used in combination with `type: "json"` */
  data: PropTypes.array,
  /** Used in combination with `type: "cypher"` */
  query: PropTypes.string,
  title: PropTypes.string,
  /** Render with the y axis to always scale to value set */
  setYAxis: PropTypes.string
};

export default Chart;
