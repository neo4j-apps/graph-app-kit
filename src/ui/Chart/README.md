Render data in a Chart visualization

Static data (XY)
```jsx
<Chart
  data={[{ x: 1, y: "116" }, { x: 2, y: "145" }, { x: 3, y: "157" }, { x: 4, y: "64" }, { x: 5, y: "152" }]}
  title="Static xy data"
  chartType="area"
  type="json"
  setYAxis="200"
/>
```
Static data (curcular):
```jsx
<Chart
  data={
    [
      { label: "Used", value: 30},
      { label: "Free", value: 20}
    ]
  }
  title="Static circular data"
  chartType="doughnut"
  type="json"
/>
```

By executing Cypher (XY):
```jsx
const stubCypherResult = {
  records: [
    {
      get: () => Math.floor((Math.random() * 100) + 1),
      keys: () => null
    }
  ]
};

<DriverProvider driver={resolvingDriver(3, stubCypherResult)}>
  <Chart
    query='RETURN rand() as n'
    title="Data from db"
    chartType="bar"
    type="cypher"
  />
</DriverProvider>
```
