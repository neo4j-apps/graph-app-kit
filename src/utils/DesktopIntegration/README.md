A React component that connects to the Neo4j Desktop API and enabled you to attach properties to it to act on events.  
See API spec to see the signature for event listener functions and for available events.  

If event type is `GRAPH_ACTIVE`, the property to attach to the component is `onGraphActive`. 
Same pattern goes for all event types.

```javascript static
<DesktopIntegration
  integrationPoint={neo4jDesktopApi}
  onMount={mountFn} // Will be called once on component mount
  onGraphActive={activeDBFn} // Will be called on ACTIVE_GRAPH events
  onDatabaseCreated={dbCreatedFn} // Will be called on DATABASE_CREATED events
/>
```
