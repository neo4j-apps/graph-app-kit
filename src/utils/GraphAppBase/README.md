This component stitches together the `<DriverProvider />`, `<DesktopIntegration />` and connection handling functionality into one easy-to-use component.

In addition to keeping track of the connection status and reconnect on graph switches this component also adds a `driver` object in application context so it's reachable from all components.
With this object you can create sessions and send Cypher to be executed on the graph. 
Components like the `<Cypher>` component uses this driver from the context to run queries and provide you with the result.  
See the `<DriverProvider>` component for more info on this.

It also provides the ability to listen on events coming from the `neo4jDesktopApi` via the `on` function passed to the render prop function.

You pass in your application to this components render property function which leaves you in total control your apps behaviour, functionality and looks.  

### Usage

```javascript static
const neo4j = require("neo4j-driver/lib/browser/neo4j-web.min.js").v1;
const integrationPoint = window.neo4jDesktopApi

const MyApp = () => <h2>This is my app!</h2>

const App = () => (
  <GraphAppBase
    driverFactory={neo4j}
    render={({ connectionState, connectionDetails, setCredentials, on, off, initialDesktopContext }) => (
      <MyApp />
    )}
    integrationPoint={integrationPoint}
  />
);

ReactDOM.render(<App />, document.getElementById('root'))
```

### Username + Password prompt
For convienience reasons there's a credentials form included in this component which can be used to collect username + password from the user in the cases where Neo4j Desktop does not know about it.  
Here's example usage code for it:

```javascript static
import { GraphAppBase, ConnectModal, CONNECTED } from "graph-app-kit/utils/GraphAppBase"

const neo4j = require("neo4j-driver/lib/browser/neo4j-web.min.js").v1;
const integrationPoint = window.neo4jDesktopApi;

const MyApp = ({ data }) => {
  return <p>Hello</p>;
};

export const App = () => (
  <GraphAppBase
    driverFactory={neo4j}
    integrationPoint={integrationPoint}
    render={({
      connectionState,
      connectionDetails,
      setCredentials,
      on,
      off,
      initialDesktopContext
    }) => {
      return [
        <ConnectModal
          key="modal"
          errorMsg={connectionDetails ? connectionDetails.message : ""}
          onSubmit={setCredentials}
          show={connectionState !== CONNECTED}
        />,
        <MyApp key="app" data={initialDesktopContext} />
      ];
    }}
  />
);

ReactDOM.render(<App />, document.getElementById('root'))
```

### Render property function signature
The render prop function provides one argument object which has these properties:

```javascript static

// This is the render prop function signature:
function render({
  connectionState: connectionState, 
  connectionDetails: connectionDetails,
  setCredentials: setCredentials,
  on: on,
  off: off,
  initialDesktopContext: initialDesktopContext
})

// Type definitions
// The state of the connection reachable in the apps context
type connectionState = 
  'CONNECTED' |
  'DISCONNECTED'

// If there's an connection error, the error object will be here
const connectionDetails = {
  message: string
} | null

// Pass connection handling prt of component what the connection credentials are.
// Useful for cases where the Desktop app passes the wrong (or none) credentials the graph apps.
function setCredentials(username: string, password: string): void {
}

// Declaration of function to be used further down in these docs
// This is not exposed by itself
function eventCallback(event: {type: string}, newContext: {}, oldContext): void {
}

// Add listeners for neo4jDesktopApi events. See the neo4jDesktopApi spec for events that
// are available.
function on(eventType: string, cb: eventCallback) {
}

// Remove neo4jDesktopApi listener
function off(eventType: string, cb: eventCallback) {
}

// Object containing data about the current Neo4j Desktop environment, including all projects etc.
// See neo4jDesktopApi spec for a complete type definition of this.
const initialDesktopContext = {
  global: {
    settings: {

    }
  },
  projects: [
    { graphs: [], id: string, name: string }
  ]
}

```
