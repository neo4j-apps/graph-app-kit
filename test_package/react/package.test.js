import React from "react";
import TestRenderer from "react-test-renderer";

// Package exports components/xxx
import { Render } from "../../dist/components/Render";
import { AsciiTable } from "../../dist/components/AsciiTable";
import { Chart } from "../../dist/components/Chart";
import { Cypher } from "../../dist/components/Cypher";
import { DesktopIntegration } from "../../dist/components/DesktopIntegration";
import { DriverProvider } from "../../dist/components/DriverProvider";
import { CypherEditor } from "../../dist/components/Editor";
import { GraphAppBase } from "../../dist/components/GraphAppBase";

// Package exports components/
import {
  Render as RenderC,
  AsciiTable as AsciiTableC,
  Chart as ChartC,
  Cypher as CypherC,
  DesktopIntegration as DesktopIntegrationC,
  DriverProvider as DriverProviderC,
  CypherEditor as CypherEditorC,
  GraphAppBase as GraphAppBaseC
} from "../../dist/components";

// Package exports lib/
import { shallowEqual } from "../../dist/lib/utils";
import { resultHasRows } from "../../dist/lib/boltTransforms";

// components
test("Render works", () => {
  const cs = [
    <Render if={true}>
      <p>Hello</p>
    </Render>,
    <RenderC if={true}>
      <p>Hello</p>
    </RenderC>
  ];

  cs.forEach(c => {
    const out = TestRenderer.create(c);
    expect(out.toJSON()).toMatchSnapshot();
  });
});
test("AsciiTable works", () => {
  const data = [["x"], ["y"]];
  const cs = [<AsciiTable data={data} />, <AsciiTableC data={data} />];

  cs.forEach(c => {
    const out = TestRenderer.create(c);
    expect(out.toJSON()).toMatchSnapshot();
  });
});
test("Chart works", () => {
  const data = [{ label: "Used", value: 30 }, { label: "Free", value: 20 }];
  const cs = [
    <Chart
      data={data}
      title="Static circular data"
      chartType="doughnut"
      type="json"
    />,
    <ChartC
      data={data}
      title="Static circular data"
      chartType="doughnut"
      type="json"
    />
  ];
  cs.forEach(c => {
    const out = TestRenderer.create(c);
    expect(out.toJSON()).toMatchSnapshot();
  });
});
test("CypherEditor works", () => {
  const cs = [<CypherEditor />, <CypherEditorC />];
  cs.forEach(c => {
    const out = TestRenderer.create(c);
    expect(out.toJSON()).toMatchSnapshot();
  });
});
test("Cypher works", () => {
  const cs = [
    <Cypher
      driver={resolvingDriver(0, "no!")}
      query="RETURN rand() as n"
      render={({ pending, error, result, tick }) => {
        return pending ? "pending" : error ? error.message : result;
      }}
    />,
    <CypherC
      driver={resolvingDriver(0, "no!")}
      query="RETURN rand() as n"
      render={({ pending, error, result, tick }) => {
        return pending ? "pending" : error ? error.message : result;
      }}
    />
  ];

  cs.forEach(c => {
    const out = TestRenderer.create(c);
    expect(out.toJSON()).toMatchSnapshot();
  });
});
test("DesktopIntegration works", () => {
  const cs = [
    <DesktopIntegration integrationPoint={true} />,
    <DesktopIntegrationC integrationPoint={true} />
  ];

  cs.forEach(c => {
    const out = TestRenderer.create(c);
    expect(out.toJSON()).toMatchSnapshot();
  });
});
test("DriverProvider works", () => {
  const cs = [
    <DriverProvider driver={resolvingDriver(0, "yes")}>Hello</DriverProvider>,
    <DriverProviderC driver={resolvingDriver(0, "yes")}>Hello</DriverProviderC>
  ];

  cs.forEach(c => {
    const out = TestRenderer.create(c);
    expect(out.toJSON()).toMatchSnapshot();
  });
});
test("GraphAppBase works", () => {
  const cs = [
    <GraphAppBase
      integrationPoint={null}
      render={() => "Hello"}
      driverFactory={{ driver: () => resolvingDriver(0, "yes") }}
    />,
    <GraphAppBaseC
      integrationPoint={null}
      render={() => "Hello"}
      driverFactory={{ driver: () => resolvingDriver(0, "yes") }}
    />
  ];

  cs.forEach(c => {
    const out = TestRenderer.create(c);
    expect(out.toJSON()).toMatchSnapshot();
  });
});

// Utils
test("shallowEqual works", () => {
  expect(shallowEqual({ x: 1 }, { x: 1 })).toBeTruthy();
});
test("resultHasRows works", () => {
  expect(resultHasRows({ result: { records: [{ x: 1 }] } })).toBeTruthy();
});

// Helpers

const sleep = (secs, shouldResolve = true, message) =>
  new Promise((resolve, reject) => {
    setTimeout(
      () => (shouldResolve ? resolve(message) : reject(new Error(message))),
      secs * 1000
    );
  });

const driver = (shouldResolve = true, waitSecs = 0, message = "") => ({
  session: () => ({
    close: () => {},
    run: () => sleep(waitSecs, shouldResolve, message)
  })
});

const resolvingDriver = (waitSecs = 0, resolveTo = "Resolved") =>
  driver(true, waitSecs, resolveTo);
