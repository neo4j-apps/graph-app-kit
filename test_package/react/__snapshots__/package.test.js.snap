// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsciiTable works 1`] = `
<pre>
  ╒═══╕
│"x"│
╞═══╡
│"y"│
└───┘
</pre>
`;

exports[`AsciiTable works 2`] = `
<pre>
  ╒═══╕
│"x"│
╞═══╡
│"y"│
└───┘
</pre>
`;

exports[`Chart works 1`] = `
<div>
  <div>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        className={null}
        style={
          Object {
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <svg
          aria-label="This is a radial-chart chart of..."
          height={500}
          role="img"
          width={500}
        >
          <g
            className="cx-group"
            transform="translate(300, 200)"
          >
            <g>
              <g
                className="cx-group vx-arcs-group"
                transform="translate(0, 0)"
              >
                <g>
                  <path
                    className="vx-arc"
                    d="M6.661338147750939e-15,-116.96153213770758A3,3,0,0,1,3.0769230769230838,-119.96054578226418A120,120,0,1,1,-68.02175660815573,98.85869019939938A3,3,0,0,1,-68.74826367607669,94.62386718753697L-42.87207476857917,59.00834860456068A3,3,0,0,1,-38.78289952029867,58.27423705891324A70,70,0,1,0,2.8767123287671192,-69.94086449406755A3,3,0,0,1,3.552713678800501e-15,-72.93833011524188Z"
                    fill="#15aabf"
                    fillOpacity={1}
                    onMouseLeave={[Function]}
                    onMouseMove={[Function]}
                    stroke="#fff"
                    strokeWidth={1}
                  />
                </g>
                <g>
                  <path
                    className="vx-arc"
                    d="M-68.74826367607668,94.62386718753697A3,3,0,0,1,-73.00032272738618,95.24155018529186A120,120,0,0,1,-3.0769230769231024,-119.96054578226418A3,3,0,0,1,-2.1316282072803006e-14,-116.96153213770758L-1.2878587085651816e-14,-72.93833011524188A3,3,0,0,1,-2.8767123287671357,-69.94086449406755A70,70,0,0,0,-43.437517844099716,54.89245889503875A3,3,0,0,1,-42.87207476857916,59.00834860456069Z"
                    fill="#fab005"
                    fillOpacity={1}
                    onMouseLeave={[Function]}
                    onMouseMove={[Function]}
                    stroke="#fff"
                    strokeWidth={1}
                  />
                </g>
              </g>
              <g
                className="cx-group vx-arcs-group"
                transform="translate(0, 0)"
              >
                <g>
                  <path
                    className="vx-arc"
                    d="M9.184850993605149e-15,-150A150,150,0,1,1,-88.167787843871,121.35254915624209A150,150,0,1,0,9.184850993605149e-15,-150Z"
                    fill="none"
                    fillOpacity={0}
                    stroke="none"
                    strokeWidth={0}
                  />
                  <text
                    fill="#495057"
                    fontFamily="BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif"
                    fontSize={12}
                    fontWeight={700}
                    letterSpacing={0.4}
                    paintOrder="stroke"
                    pointerEvents="none"
                    stroke="none"
                    textAnchor="middle"
                    x={142.65847744427302}
                    y={46.35254915624214}
                  >
                    60.00%
                  </text>
                </g>
                <g>
                  <path
                    className="vx-arc"
                    d="M-88.167787843871,121.35254915624209A150,150,0,0,1,-2.7554552980815446e-14,-150A150,150,0,0,0,-88.167787843871,121.35254915624209Z"
                    fill="none"
                    fillOpacity={0}
                    stroke="none"
                    strokeWidth={0}
                  />
                  <text
                    fill="#495057"
                    fontFamily="BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif"
                    fontSize={12}
                    fontWeight={700}
                    letterSpacing={0.4}
                    paintOrder="stroke"
                    pointerEvents="none"
                    stroke="none"
                    textAnchor="middle"
                    x={-142.65847744427305}
                    y={-46.35254915624209}
                  >
                    40.00%
                  </text>
                </g>
              </g>
            </g>
          </g>
        </svg>
        <style
          type="text/css"
        >
          
      .vx-arc:hover {
        opacity: 0.7;
      }
    
        </style>
      </div>
    </div>
    <div
      className="vx-legend"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <div
        className="vx-legend-item"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "margin": "0",
          }
        }
      >
        <div
          className="vx-legend-shape"
          style={
            Object {
              "display": "flex",
              "height": "#15aabf",
              "margin": "2px 4px 2px 0",
              "width": "#15aabf",
            }
          }
        >
          <div
            style={
              Object {
                "background": "#15aabf",
                "height": 15,
                "style": undefined,
                "width": 15,
              }
            }
          />
        </div>
        <div
          className="vx-legend-label"
          style={
            Object {
              "display": "flex",
              "flex": "1",
              "justifyContent": "left",
              "margin": "0 4px",
            }
          }
        >
          Used
        </div>
      </div>
      <div
        className="vx-legend-item"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "margin": "0",
          }
        }
      >
        <div
          className="vx-legend-shape"
          style={
            Object {
              "display": "flex",
              "height": "#fab005",
              "margin": "2px 4px 2px 0",
              "width": "#fab005",
            }
          }
        >
          <div
            style={
              Object {
                "background": "#fab005",
                "height": 15,
                "style": undefined,
                "width": 15,
              }
            }
          />
        </div>
        <div
          className="vx-legend-label"
          style={
            Object {
              "display": "flex",
              "flex": "1",
              "justifyContent": "left",
              "margin": "0 4px",
            }
          }
        >
          Free
        </div>
      </div>
    </div>
  </div>
  <h2>
    Static circular data
  </h2>
</div>
`;

exports[`Chart works 2`] = `
<div>
  <div>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        className={null}
        style={
          Object {
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <svg
          aria-label="This is a radial-chart chart of..."
          height={500}
          role="img"
          width={500}
        >
          <g
            className="cx-group"
            transform="translate(300, 200)"
          >
            <g>
              <g
                className="cx-group vx-arcs-group"
                transform="translate(0, 0)"
              >
                <g>
                  <path
                    className="vx-arc"
                    d="M6.661338147750939e-15,-116.96153213770758A3,3,0,0,1,3.0769230769230838,-119.96054578226418A120,120,0,1,1,-68.02175660815573,98.85869019939938A3,3,0,0,1,-68.74826367607669,94.62386718753697L-42.87207476857917,59.00834860456068A3,3,0,0,1,-38.78289952029867,58.27423705891324A70,70,0,1,0,2.8767123287671192,-69.94086449406755A3,3,0,0,1,3.552713678800501e-15,-72.93833011524188Z"
                    fill="#15aabf"
                    fillOpacity={1}
                    onMouseLeave={[Function]}
                    onMouseMove={[Function]}
                    stroke="#fff"
                    strokeWidth={1}
                  />
                </g>
                <g>
                  <path
                    className="vx-arc"
                    d="M-68.74826367607668,94.62386718753697A3,3,0,0,1,-73.00032272738618,95.24155018529186A120,120,0,0,1,-3.0769230769231024,-119.96054578226418A3,3,0,0,1,-2.1316282072803006e-14,-116.96153213770758L-1.2878587085651816e-14,-72.93833011524188A3,3,0,0,1,-2.8767123287671357,-69.94086449406755A70,70,0,0,0,-43.437517844099716,54.89245889503875A3,3,0,0,1,-42.87207476857916,59.00834860456069Z"
                    fill="#fab005"
                    fillOpacity={1}
                    onMouseLeave={[Function]}
                    onMouseMove={[Function]}
                    stroke="#fff"
                    strokeWidth={1}
                  />
                </g>
              </g>
              <g
                className="cx-group vx-arcs-group"
                transform="translate(0, 0)"
              >
                <g>
                  <path
                    className="vx-arc"
                    d="M9.184850993605149e-15,-150A150,150,0,1,1,-88.167787843871,121.35254915624209A150,150,0,1,0,9.184850993605149e-15,-150Z"
                    fill="none"
                    fillOpacity={0}
                    stroke="none"
                    strokeWidth={0}
                  />
                  <text
                    fill="#495057"
                    fontFamily="BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif"
                    fontSize={12}
                    fontWeight={700}
                    letterSpacing={0.4}
                    paintOrder="stroke"
                    pointerEvents="none"
                    stroke="none"
                    textAnchor="middle"
                    x={142.65847744427302}
                    y={46.35254915624214}
                  >
                    60.00%
                  </text>
                </g>
                <g>
                  <path
                    className="vx-arc"
                    d="M-88.167787843871,121.35254915624209A150,150,0,0,1,-2.7554552980815446e-14,-150A150,150,0,0,0,-88.167787843871,121.35254915624209Z"
                    fill="none"
                    fillOpacity={0}
                    stroke="none"
                    strokeWidth={0}
                  />
                  <text
                    fill="#495057"
                    fontFamily="BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif"
                    fontSize={12}
                    fontWeight={700}
                    letterSpacing={0.4}
                    paintOrder="stroke"
                    pointerEvents="none"
                    stroke="none"
                    textAnchor="middle"
                    x={-142.65847744427305}
                    y={-46.35254915624209}
                  >
                    40.00%
                  </text>
                </g>
              </g>
            </g>
          </g>
        </svg>
        <style
          type="text/css"
        >
          
      .vx-arc:hover {
        opacity: 0.7;
      }
    
        </style>
      </div>
    </div>
    <div
      className="vx-legend"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <div
        className="vx-legend-item"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "margin": "0",
          }
        }
      >
        <div
          className="vx-legend-shape"
          style={
            Object {
              "display": "flex",
              "height": "#15aabf",
              "margin": "2px 4px 2px 0",
              "width": "#15aabf",
            }
          }
        >
          <div
            style={
              Object {
                "background": "#15aabf",
                "height": 15,
                "style": undefined,
                "width": 15,
              }
            }
          />
        </div>
        <div
          className="vx-legend-label"
          style={
            Object {
              "display": "flex",
              "flex": "1",
              "justifyContent": "left",
              "margin": "0 4px",
            }
          }
        >
          Used
        </div>
      </div>
      <div
        className="vx-legend-item"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "margin": "0",
          }
        }
      >
        <div
          className="vx-legend-shape"
          style={
            Object {
              "display": "flex",
              "height": "#fab005",
              "margin": "2px 4px 2px 0",
              "width": "#fab005",
            }
          }
        >
          <div
            style={
              Object {
                "background": "#fab005",
                "height": 15,
                "style": undefined,
                "width": 15,
              }
            }
          />
        </div>
        <div
          className="vx-legend-label"
          style={
            Object {
              "display": "flex",
              "flex": "1",
              "justifyContent": "left",
              "margin": "0 4px",
            }
          }
        >
          Free
        </div>
      </div>
    </div>
  </div>
  <h2>
    Static circular data
  </h2>
</div>
`;

exports[`Cypher works 1`] = `"pending"`;

exports[`Cypher works 2`] = `"pending"`;

exports[`CypherEditor works 1`] = `
<div
  className="ReactCodeMirror"
/>
`;

exports[`CypherEditor works 2`] = `
<div
  className="ReactCodeMirror"
/>
`;

exports[`DesktopIntegration works 1`] = `null`;

exports[`DesktopIntegration works 2`] = `null`;

exports[`DriverProvider works 1`] = `"Hello"`;

exports[`DriverProvider works 2`] = `"Hello"`;

exports[`GraphAppBase works 1`] = `
<div>
  Hello
</div>
`;

exports[`GraphAppBase works 2`] = `
<div>
  Hello
</div>
`;

exports[`Render works 1`] = `
<p>
  Hello
</p>
`;

exports[`Render works 2`] = `
<p>
  Hello
</p>
`;
